File: backend/app/schemas/conversation.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict


class ConversationCreate(BaseModel):
    user_id: int


class ConversationResponse(BaseModel):
    id: int

    user1_id: int
    user2_id: int

    created_at: datetime
    updated_at: datetime

    model_config = ConfigDict(from_attributes=True)


class ConversationListResponse(BaseModel):
    conversations: list[ConversationResponse]

    model_config = ConfigDict(from_attributes=True)

File: frontend/src/types/conversation.ts
export interface Conversation {
    id: number;

    user1_id: number;

    user2_id: number;

    created_at: string;

    updated_at: string;

    user1?: {
        id: number;
        username: string;
        avatar_url?: string;
    };

    user2?: {
        id: number;
        username: string;
        avatar_url?: string;
    };
}

export interface ConversationListResponse {
    conversations: Conversation[];
}